api/src/main/java/org/apache/iceberg/types/TypeUtil.java

```diff
@@ -709,6 +709,10 @@ public T map(Types.MapType map, Supplier<T> keyResult, Supplier<T> valueResult)
       return null;
     }
 
+    public T variant() {
+      return null;
```

Comment:
Would it be safer to throw `UnsupportedOperationException` here instead? I think that's what we have done for expression visitors when we add new cases.

Reply:
I can see others like map()/list() all return null here.